SimConnection.py

```diff
@@ -19,6 +19,13 @@ class SimPacketId(Enum):
 
 
 class SimConnection(IConnection):
+    def _rocket_handshake(self):
+        assert self.rocket.stdout.read(3) == b"SYN"
```

Comment:
Just for consistency's sake, let's use `self.stdout` instead of `self.rocket.stdout`. Unless there's a reason for doing it this way that I'm not aware of.

Reply:
Not actually sure what StreamLogger does so I opted to not touch it. Only reason I'd note is to keep self._rocket_handshake() as close to Popen as possible, which would put it in front of the setting up of self.stdout

Reply:
It has a circular buffer of the read history. It can be very helpful for debugging.

Reply:
Especially now that you moved the handshake call below the creation of the StreamLogger, we should use it lol